### DISCUSSION TITLE
get request object in filter

### DISCUSSION THREAD

**Discussion by 3hson** [mikro-orm/mikro-orm#4607] [Q&A] [answered]
I want to filter by user id from request object.
Is it possible to get request object in filter decorator?

```ts
@Filter({
  name: 'owned',
  default: true,
  args: false,
  cond: async () => {
    // get authUser from req object
    return {
      user: authUser.id,
    };
  },
})
```

**Comment by B4nan** [comment, marked as answer]
You can set the filter parameters on EM forks (e.g. in a middleware).

https://mikro-orm.io/docs/filters#parameters
https://mikro-orm.io/docs/filters#global-filters
